src/Symfony/Component/Validator/ConstraintValidator.php

```diff
@@ -88,6 +88,14 @@ protected function formatValue(mixed $value, int $format = 0): string
             return $value->format('Y-m-d H:i:s');
         }
 
+        if ($value instanceof \UnitEnum) {
+            if ($format & self::OBJECT_TO_STRING) {
```

Comment:
I would remove this check personally, I don't see it's value.

Reply:
The first it's to check if it's an enum instead of a generic object.  The second is to maintain the same behavior as with objects, if the flag is set you return a value if not set, you return the string of what it is, an "enum" in this case and "object" in the object case.

I think we should keep them to maintain a similar functionality as with objects.

Reply:
It's true that it could be better to have another flag like `self::NAME_OF_ENUM` specific for the enums (the same way objects have `self::OBJECT_TO_STRING` and `self::PRETTY_DATE` for dates. As this is my first PR I wanted to change the less possible but I can add it if it fits.

Reply:
No need for any flag IMHO.

Reply:
We display strings and numbers, so why not enums? I see each flag as a complication. The class of the enum is already exposed, not saying which case is concerned just worsen DX to me, this is not something that devs want to opt-in for, nor disable.

Reply:
Ok, I will remove the flag and return always the name